discovery-utils/src/discovery/mod.rs

```diff
@@ -0,0 +1,201 @@
+pub mod v0;
+
+/// Path of the Agent registration socket
+pub const AGENT_REGISTRATION_SOCKET: &str = "/var/lib/akri/agent-registration.sock";
```

Comment:
what would happen if multiple agents were running on a single node?

Reply:
I am not sure. I would think that the first one would get the socket. Thats not a scenario we want to support, right? Just like you wouldnt ever have 2 kubelet device plugin managers running on a node.

Reply:
definitely not something we want.  but something i could see happening:

1. accidentally apply 2 daemonsets of our agent
2. create an agent with embedded opcua and a second agent with embedded udev .. apply both

i wonder if there is a way to enforce an "only one agent pod can run on a node at a time" policy ... either at the k8s level or in our code somehow.

Reply:
I think it would be hard to do that by accident since we dont allow you to configure the name of the agent. If you try to install a second agent you get the error:
```
Error: rendered manifests contain a resource that already exists. Unable to continue with install: DaemonSet "akri-agent-daemonset" in namespace "default" exists and cannot be imported into the current release
```
If you create another namespace and install the agent, then, the second agent will panic because it cannot use the agent registration socket:
```
internal_run_registration_server - registration server listening on socket /var/lib/akri/agent-registration.sock
error binding to 0.0.0.0:8080: error creating server listener: Address already in use (os error 98)', [2021-03-02T01:06:15Z TRACE akri_shared::k8s] Loading in-cluster config
```
So long as people are using our helm chart to install the agent and not trying to install it in multiple namespaces, we will be fine. I dont see not abiding by these as likely.